### ISSUE TITLE
NEW COMMAND FOUND

### ISSUE THREAD

**Issue by DaveGut** [bacl/WAM_API_DOC#1]
I finally broke some of the code.  Following information is very interesting:

Playing a URL on the Speakers and Older (non SoundPlus) Soundbars. This is used in playTrackAndResume, playTrackAndRestore, and playTrack in the SmartThings.  It is a NEW command to the lexicon.

/UIC?cmd=<pwron>on</pwron><name>SetUrlPlayback</name>
<p type="cdata" name="url" val="empty"><![CDATA[${urlTrack}]]>
</p><p type="dec" name="buffersize" val="0"/>
<p type="dec" name="seektime" val="0"/>
<p type="dec" name="resume" val="${resume}"/>

Variables:  ${urlTrack} is a text string, no quotes.  Example:  http://s3.amazonaws.com/smartapp-media/sonos/dogs.mp3
	${resume} is 0 or 1 (with quotes).  0 is do not resume.  1 is resume playing.

Playing a URL on the SoundPlus (newer) Soundbars:  Use the port 9797 DMR / AVTransport UPnP commands.

Software versions:  

There are two software versions for the Speakers and Soundbars.  They are different

Speakers and non-SoundPlus Soundbars:  Current version 3117.1 / 181R
a.	Does not have the UPnP DMR capability (which includes AVTransport).
b.	Uses “SetUrlPlayback” for URL play.

SoundPlus (Sound+) Soundbars (i.e., HS-MS650):  Current version 1146.4 / 196R
a.	 Supports the UPnP DMR capability (which includes AVTransport).
b.	Does not have the SetUrlPlayback command.

**Comment by bacl** [comment]
That's great news @DaveGut !!
I have tested it and it works!
But It doesn't react to the player controls  (ie pause, stop...)  even  from the hardware ones. 
Through SmartThings were you able to stop or pause the music stream?

**Comment by DaveGut** [comment]
No assumptions from me.  Make sure you are trying the UIC and CPM versions of the pause.

From my side, I will see what SmartThings sends to the speaker on playTrack command.  I think the same.  I will also see if I can pause or stop on my system.

**Comment by DaveGut** [comment]
I just completed a check on SmartThings using my Device Handler (slightly modified) and the short “dogs barking” audio notification.  The device was an R1 Speaker with software 3117.1 / 181R.

Using the UIC version of play  / pause (below), I successful paused the audio and then restarted the audio.

/UIC?cmd=%3Cpwron%3Eon%3C/pwron%3E%3Cname%3ESetPlaybackControl%3C/name%3E%3Cp%20type=%22str%22%20name=%22playbackcontrol%22%20val=%22${playbackControl}%22/%3E



From: David Gutheinz
Sent: Saturday, May 5, 2018 12:05 AM
To: 'bacl/WAM_API_DOC' <reply@reply.github.com>; bacl/WAM_API_DOC <WAM_API_DOC@noreply.github.com>
Cc: David Gutheinz <djgutheinz@outlook.com>; Mention <mention@noreply.github.com>
Subject: RE: [bacl/WAM_API_DOC] NEW COMMAND FOUND (#1)

No assumptions from me.  Make sure you are trying the UIC and CPM versions of the pause.

From my side, I will see what SmartThings sends to the speaker on playTrack command.  I think the same.  I will also see if I can pause or stop on my system.

**Comment by bacl** [comment]
Thank you for taking the time to check that. 

well, this was weird... 

I all the commands that I sent to control the player didn't work, so I restarted the speaker (R1 3117.1 / 181R) but still didnt respond.
Switched for spotify and them sent the command SetUrlPlayback and still no response from player controls.
Changed modes to bluetooth but the music from setUrl was not stopped and kept playing in the background...
Went back to wifi mode and tried again  setUrl and still no response...
tried playing music from dlna and them from tunein  and tried again  setUrl and now it finally responded! 

Don't know what to think of it, at least now it's back to normal operation by responding to all of the commands.
